### ISSUE TITLE
Some question about state tx infer

### ISSUE THREAD

**Issue by raphaelmourad** [ArcInstitute/state#187]
Hi,

after carefully reading the paper, I understood that the "transformer backbone fST models perturbation effects across the cell sets", and that the loss is computed between the distributions of cells. Therefore, information is shared among cells in the batch (here I am talking about the batch of the neural network), meaning that cells are not independently modeled at all. 

Therefore, this implies that when someone wants to predict the effect of a drug on a set of cells from a tissue, state tx infer must be applied on a set of cells from the same cell type (and therefore that cell type inference must be done before). If cells from different cell types are mixed, then state tx infer will mix information from cells from different cell types and this must impact significantly predictions.

I have done both in practice:
1) state tx infer from a set of cells from the same inferred cell type X (inferred using cell typist)
2)  state tx infer from a whole scRNA-seq experiment mixing cells from different cell types, then take only results from cells from cell type X
Both approaches give very different results. In the approach 1, I obtain drug effects that much less significant than using approach 2.

I believe the approach 2 is incorrect, but I would need validation from the authors if possible!!

Regards
Raphaël

**Comment by liu-wq** [comment]
Hi Raphaël,

I have the same problem as you. Have you done the UMAP plot for approach 2? In my case, approach 2 will result in the loss of cell type identity. PCA-UMP was performed using the predicted expression matrix, and all cells were eventually clustered together. 

Thanks,
Leo

**Comment by raphaelmourad** [comment]
Indeed, UMAP plot from approach 2 mixes the cell types (not completely but still a lot!).

**Comment by abhinadduri** [comment]
Hi @raphaelmourad and @liu-wq , thanks for your interest and  for making the issue. We found a bug in the previous version of the infer code, where cell sets were not homogeneous according to cell type or cell line. You can test the new version in this branch: aadduri/infer_groupby

Note this version introduces an API changes, which is why I have been hestitant to merge it. the infer script will now take an existing adata, and replace all cells with predictions. this means no more preprocessing for the infer command.


For your actual question, my intuition is that the model should do better in setting 1, as this is how the model is trained. We tested what you describe as setting two (Fig S9, Panel D), and the model self-attention seemed to by cell-type-specific there. Can you share more information on your inference performance - very curious to hear more about your use case.

**Comment by raphaelmourad** [comment]
The strongest effect i got when i mixed cells from different cell types (as compared to when i didn't) and then used state tx infer was that many more genes got significantly differential expression when tested by wilcoxon test from scanpy (i compared dmso prediction with drug x prediction). Pvalues can go from 0.1 to 1x-20.

**Comment by liu-wq** [comment]
Hi @abhinadduri,
Because Tahoe-100M only includes the drugs relevant to my project, I input the scRNA-seq data of the human postmortem brain into STATE to evaluate drug effects. However, in the predicted gene expression, the marker genes lost their cell-type specificity, and the UMAP collapsed into a single cluster.
